stdlib/REPL/src/REPLCompletions.jl

```diff
@@ -136,6 +136,7 @@ end
 
 function filtered_mod_names(ffunc::Function, mod::Module, name::AbstractString, all::Bool = false, imported::Bool = false)
     ssyms = names(mod, all = all, imported = imported)
+    all || filter!(Base.Fix1(Base.isexported, mod), ssyms)
```

Comment:
What about making this an option on `names`. Seems useful?

Reply:
We considered that for a bit [here](https://github.com/JuliaLang/julia/pull/50105#discussion_r1291914524), but I ended up thinking that the predicates we use to filter the names are awfully complicated and hard to understand and accurately document as is:

https://github.com/JuliaLang/julia/blob/ebe1a37af57cb472101d6ede43329ea5ef2e0138/src/module.c#L955-L959

and that it would be better to give users access to a full list of names and a set of predicates to filter with themselves. This arises from the belief that complex configuration is better expressed with code than keywords.

(I'm merging this bugfix now, but that does not close this discussion, feel free to open a feature request issue)